yt_dlp/extractor/echo360.py

```diff
@@ -0,0 +1,129 @@
+        if video_id is None:
+            raise ExtractorError('Video id was not found')
+        query_strings = traverse_obj(video, ('sourceQueryStrings', 'queryStrings')) or []
+        duration = float_or_none(self._search_regex(
+            r'PT(\d+\.?\d+)S', traverse_obj(video, ('playableAudioVideo', 'duration')),
+            'video duration', default=None, fatal=False))
+
+        formats = []
+        for track in variadic(traverse_obj(video, ('playableAudioVideo', 'playableMedias')) or []):
+            href = track.get('uri')
+            if href is None:
+                continue
+            href = self._update_url_query(href, self._get_query_string(href, query_strings))
+            if track.get('isHls') or determine_ext(href, None) == 'm3u8':
+                hls_formats = self._extract_m3u8_formats(
+                    href, video_id, live=track.get('isLive'), m3u8_id='hls', entry_protocol='m3u8_native', fatal=False
+                )
+
+                for hls_format in hls_formats:
+                    query_string = self._get_query_string(hls_format['url'], query_strings)
+                    hls_format['extra_param_to_segment_url'] = query_string
+                    hls_format['url'] = self._update_url_query(hls_format['url'], query_string)
+
+                formats.extend(hls_formats)
+
+        return {
+            'id': video_id,
+            'formats': formats,
+            'title': video.get('mediaName'),
+            'duration': duration,
+        }
+
+
+class Echo360IE(Echo360BaseIE):
+    _VALID_URL = rf'''(?x)
+        https?://(?P<host>{Echo360BaseIE._INSTANCES_RE})
+        /media/(?P<id>{Echo360BaseIE._UUID_RE})/public'''
+
+    _API_BASE = 'https://%s/api/ui/echoplayer/public-links/%s/media/%s/player-properties'
+
+    _TESTS = [
+        {
+            'url': 'https://echo360.org.uk/media/1d8392aa-a3e7-4e78-94cf-b6532c27208c/public',
+            'info_dict': {
+                'id': '3c7ae6e0-fa19-432d-aa21-c283b4276f2a',
+                'ext': 'mp4',
+                'title': '3-4 Force + moment + mechanics.mp4',
+                'duration': 4731.888,
+            },
+            'params': {'skip_download': 'm3u8'}
+        },
+        {
+            'url': 'https://echo360.net.au/media/f04960a9-2efc-4b63-87b5-72e629081d15/public',
+            'info_dict': {
+                'id': '6098a147-2d65-40f3-b9e9-a0204afe450c',
+                'ext': 'mp4',
+                'title': 'EXSC634_Online_Workshop_Week_4.mp4',
+                'duration': 6659.72,
+            },
+            'params': {'skip_download': 'm3u8'}
+        },
+    ]
+
+    def _real_extract(self, url):
+        host, video_id = self._match_valid_url(url).group('host', 'id')
+        webpage = self._download_webpage(url, video_id)
+
+        player_config = self._parse_json(self._search_regex(
+            r'Echo\["mediaPlayerBootstrapApp"\]\("({[^}]*})"\);', webpage, 'player config').replace('\\"', "\""),
+            video_id)
```

Comment:
I think you can use `self._search_json`

Reply:
It did not work for me because of the escaped quotes. Do you know if it is possible?

Reply:
The json that I want to extract looks like this: 
```
if (document.readyState === "complete") {
          Echo["mediaPlayerBootstrapApp"]("{\"isMobile\":true,\"locale\":\"de\",\"cookiesDisabledUrl\":\"https:\/\/help.echo360.com\/hc\/en-us\/articles\/360035408871-Recommended-Browsers\",\"mediaId\":\"3c7ae6e0-fa19-432d-aa21-c283b4276f2a\",\"sessionId\":\"c7e71ba1-ae17-4bc2-9e81-10f7f09b57c0\",\"publicLinkId\":\"1d8392aa-a3e7-4e78-94cf-b6532c27208c\",\"id\":\"r00841580-b31a-4963-81f2-d00737aeae89\"}");
        }
```

All quotes are escaped... not sure if there is a better way to extract it.

Reply:
```suggestion
        player_config = self._search_json(r'Echo\["mediaPlayerBootstrapApp"\]\("', webpage, 'player config',
                                          video_id, transform_source=lambda x: x.replace(R'\"', '"'))
```

Reply:
Is there a reason why you used uppercase R for the raw string?

Reply:
It's convention to use `r` for regex and `R` for other raw strings